worlds/civ_6/Regions.py

```diff
@@ -0,0 +1,178 @@
+from BaseClasses import CollectionState, LocationProgressType, Region
+from .Data import get_era_required_items_data, get_progressive_districts_data
+from .Items import format_item_name, get_item_by_civ_name
+from .Enum import CivVICheckType, EraType
+from .Locations import CivVILocation
+from .ProgressiveDistricts import get_flat_progressive_districts
+from .Options import CivVIOptions
+
+if TYPE_CHECKING:
+    from . import CivVIWorld
+    from Items import CivVIItemData
+
+
+def get_cumulative_prereqs_for_era(end_era: EraType, exclude_progressive_items: bool = True, item_table: Dict[str, 'CivVIItemData'] = None) -> List['CivVIItemData']:
+    """Gets the specific techs/civics that are required for the specified era as well as all previous eras"""
+    cumulative_prereqs = []
+    era_required_items = {}
+    era_required_items = get_era_required_items_data()
+
+    for era in EraType:
+        cumulative_prereqs += era_required_items[era.value]
+        if era == end_era:
+            break
+    # If we are excluding progressive items, we need to remove them from the list of expected items (TECH_BRONZE_WORKING won't be here since it will be PROGRESSIVE_ENCAMPMENT)
+    if exclude_progressive_items:
+        flat_progressive_items = get_flat_progressive_districts()
+        prereqs_without_progressive_items = []
+        for item in cumulative_prereqs:
+            if item in flat_progressive_items:
+                continue
+            else:
+                prereqs_without_progressive_items.append(item)
+
+        return [get_item_by_civ_name(prereq, item_table) for prereq in prereqs_without_progressive_items]
+
+    return [get_item_by_civ_name(prereq, item_table) for prereq in cumulative_prereqs]
+
+
+def has_required_progressive_districts(state: CollectionState, era: EraType, player: int) -> bool:
+    """ If player has progressive items enabled, it will count how many progressive techs it should have, otherwise return the default array"""
+    progressive_districts = get_progressive_districts_data()
+
+    item_table = state.multiworld.worlds[player].item_table
+    # Verify we can still reach non progressive items
+    all_previous_items_no_progression = get_cumulative_prereqs_for_era(
+        era, True, item_table)
+    if not state.has_all([item.name for item in all_previous_items_no_progression], player):
+        return False
+
+    # Verify we have the correct amount of progressive items
+    all_previous_items = get_cumulative_prereqs_for_era(
+        era, False, item_table)
+    required_counts: Dict[str, int] = {}
+
+    for key, value in progressive_districts.items():
+        required_counts[key] = 0
+        for item in all_previous_items:
+            if item.civ_name in value:
+                required_counts[key] += 1
+
+    for key, value in required_counts.items():
+        has_amount = state.has(format_item_name(key), player, required_counts[key])
+        if not has_amount:
+            return False
```

Comment:
```suggestion
        if not state.has(format_item_name(key), player, required_counts[key]):
            return False
```
Would recommend one-lining this